// connect_dynamodb/instructor.js
const { core } = require('./core')

module.exports = {
  get_item: async (after_school_id, instructor_id) => {
    const response = await core.query('PK = :p_key AND SK = :s_key', {
      ':p_key': 'AFTER_SCHOOL#' +  after_school_id,
      ':s_key': 'INSTRUCTOR#' + instructor_id,
    })
    return (response.Items && response.Items.length > 0) ? response.Items[0] : {}
  },
  get_all: async (after_school_id) => {
    const response = await core.query('PK = :p_key AND begins_with(SK, :s_key)', {
      ':p_key': 'AFTER_SCHOOL#' +  after_school_id,
      ':s_key': 'INSTRUCTOR#',
    })
    return response.Items
  },
  put: async (after_school_id, instructor_id, instructor_name, qualification, additional, medical_care, seiki, koyou, order) => {
    return await core.put({
      PK: 'AFTER_SCHOOL#' +  after_school_id,
      SK: 'INSTRUCTOR#' +  instructor_id,
      Name: instructor_name,
      Qualification: qualification,
      Additional: additional,
      MedicalCare: medical_care,
      Seiki: seiki,
      Koyou: koyou,
      Order: order,
    })
  },
  delete: async (after_school_id, instructor_id) => {
    return await core.delete({
      PK: 'AFTER_SCHOOL#' +  after_school_id,
      SK: 'INSTRUCTOR#' +  instructor_id,
    })
  }
}
